txnkv/transaction/txn.go

```diff
@@ -162,6 +164,9 @@ type KVTxn struct {
 	aggressiveLockingDirty   atomic.Bool
 
 	forUpdateTSChecks map[string]uint64
+
+	isPipelined    bool
```

Comment:
Is it enough to just using one field `pipelinedMemDB *unionstore.PipelinedMemDB` and `pipelinedMemDB == nil` to indicate if it's pipelined?

Reply:
Now we union the `MemDB` and `PipelinedMemDB` into `MemBuffer` interface, this field is required.